pilz_control/test/unittest_pilz_joint_trajectory_controller.cpp

```diff
@@ -434,127 +686,104 @@ TEST_P(PilzJointTrajectoryControllerIsExecutingTest, testHoldAtStart)
 
   bool is_executing_result;
   EXPECT_TRUE(invokeIsExecuting(is_executing_result));
-  EXPECT_TRUE(is_executing_result) << "Controller is not executing hold trajectory at start";
+  EXPECT_TRUE(is_executing_result) << "Failed to detect stop trajectory execution at start.";
 
-  /**********
-   * Step 2 *
-   **********/
-  ros::Duration stop_duration{STOP_TRAJECTORY_DURATION + DEFAULT_UPDATE_PERIOD_SEC};
+  ros::Duration stop_duration{STOP_TRAJECTORY_DURATION_SEC + 2*DEFAULT_UPDATE_PERIOD_SEC};
   progressInTime(stop_duration);
   updateController();
 
   EXPECT_TRUE(invokeIsExecuting(is_executing_result));
-  EXPECT_FALSE(is_executing_result) << "Controller is executing unexpectedly";
-
-  /**********
-   * Step 3 *
-   **********/
-  std_srvs::TriggerRequest req;
-  std_srvs::TriggerResponse resp;
-  EXPECT_TRUE(controller_->handleUnHoldRequest(req, resp));
-  EXPECT_TRUE(resp.success);
+  EXPECT_FALSE(is_executing_result) << "There should be no execution to detect";
+}
 
+TEST_P(PilzJointTrajectoryControllerIsExecutingTest, testNotExecutingAfterUnhold)
+{
+  ASSERT_TRUE(performFullControllerStartup());
+  
+  bool is_executing_result;
   EXPECT_TRUE(invokeIsExecuting(is_executing_result));
-  EXPECT_FALSE(is_executing_result) << "Controller is executing unexpectedly";
+  EXPECT_FALSE(is_executing_result) << "There should be no execution to detect";
 }
 
-/**
- * @brief Test if the controller starts and completes the execution of a single action goal.
- */
-TEST_P(PilzJointTrajectoryControllerIsExecutingTest, testSingleActionGoal)
+TEST_P(PilzJointTrajectoryControllerIsExecutingTest, testActionGoalExecution)
 {
   ASSERT_TRUE(performFullControllerStartup());
 
-  /**********
-   * Step 1 *
-   **********/
-  ros::Duration goal_duration{2.0};
-  GoalType goal {generateSimpleGoal(goal_duration)};
-
+  GoalType goal {generateSimpleGoal()};
   trajectory_action_client_.sendGoal(goal);
+
+  std::chrono::milliseconds movement_timeout {WAIT_FOR_MOVEMENT_STARTED_MSEC};
+  EXPECT_TRUE(updateControllerUntil([this](){ return robot_->isMoving(); }, movement_timeout));
+
   bool is_executing_result;
-  EXPECT_TRUE(waitForIsExecutingResult(true, is_executing_result));
+  EXPECT_TRUE(invokeIsExecuting(is_executing_result));
+  EXPECT_TRUE(is_executing_result) << "Failed to detect action goal execution";
 
-  /**********
-   * Step 2 *
-   **********/
   ros::Duration default_update_period {DEFAULT_UPDATE_PERIOD_SEC};
```

Comment:
I think this line is obsolete and actually clutters the test code. Simply integrate it directly in the line where the variable is used.

Reply:
Done.